src/approaches/llm_probe_approach.py

```diff
@@ -0,0 +1,137 @@
+from predicators.src.settings import CFG
+from predicators.src.structs import GroundAtom, Object, ParameterizedOption, \
+    Sequence, State, Task, _GroundNSRT
+
+
+class LLMProbeApproach(OpenLoopLLMApproach):
+    """LLMProbeApproach definition."""
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "llm_probe"
+
+    def _get_llm_based_plan(
+            self, state: State, atoms: Set[GroundAtom],
+            goal: Set[GroundAtom]) -> Optional[List[_GroundNSRT]]:
+        """Query the LLM and use the outputs to initialize the queue in A*."""
+        # To initialize the queue in A*, we will build up a partial policy
+        # and then run policy-guided A*.
+        start_time = time.time()
+        objects = set(state)
+        partial_policy_dict: Dict[FrozenSet[GroundAtom], _GroundNSRT] = {}
+        for option_plan in self._get_llm_based_option_plans(
+                atoms, objects, goal):
+            ground_nsrt_plan = self._option_plan_to_nsrt_plan(
+                option_plan, atoms, objects, goal)
+            assert ground_nsrt_plan is not None
+            # Simulate the ground NSRT plan and record the (atoms, ground NSRT)
+            # pairs encountered along the way to build up a partial policy.
+            single_partial_policy_dict = {}
+            cur_atoms = atoms
+            for ground_nsrt in ground_nsrt_plan:
+                # If we run into an invalid ground NSRT, the rest of the plan
+                # is going to be garbage, so stop here.
+                if not ground_nsrt.preconditions.issubset(cur_atoms):
+                    break
+                single_partial_policy_dict[frozenset(cur_atoms)] = ground_nsrt
+                cur_atoms = utils.apply_operator(ground_nsrt, cur_atoms)
+            # Update the overall partial policy. Note that there may be
+            # entries that get overridden. This is okay, and there's no
+            # reason to prefer one option plan over another, so we just let
+            # the overriding happen.
+            partial_policy_dict.update(single_partial_policy_dict)
+        # Run policy-guided A*. Note that we importantly want to use
+        # sesame_plan because we will want to report the metrics collected in
+        # sesame_plan.
+        abstract_policy = lambda a, o, g: partial_policy_dict.get(
+            frozenset(a), None)
+        max_policy_guided_rollout = CFG.horizon  # not important
+        nsrts = self._get_current_nsrts()
+        preds = self._get_current_predicates()
+        task = Task(state, goal)
+        options, metrics = self._run_sesame_plan(
+            task,
+            nsrts,
+            preds,
+            CFG.timeout - (time.time() - start_time),
+            CFG.seed,
+            abstract_policy=abstract_policy,
+            max_policy_guided_rollout=max_policy_guided_rollout)
+        self._save_metrics(metrics, nsrts, preds)
+        # Now convert the options back into ground NSRTs (:facepalm:). This
```

Comment:
i don't really get why this is raising alarm bells... aren't you just going from the `plan` returned by sesame back to the `skeleton`? i agree that it's non-invertible in general, but another way to do this would be to just change sesame_plan() to return the skeleton along with the plan and the metrics. do you prefer doing it that way?

Reply:
it's raising alarm bells because after we call sesame plan, we have already figured out option parameters (the continuous ones), but here we're throwing those away and counting on the metacontroller sampling to fill them back in later. this doesn't matter for us right now only because there are no continuous parameters in the PDDL envs.